### ISSUE TITLE
[BUG] Condition Tensor Always Zero Due to Incorrect Masking Logic in `forward`

### ISSUE THREAD

**Issue by romberol** [ASLP-lab/DiffRhythm#18]
File affected: https://github.com/ASLP-lab/DiffRhythm/blob/92ba3b3/model/cfm.py
## Description:
I'm trying to understand the conditioning mechanism in the `forward` method of the model. In particular, the variable `rand_span_mask` seems to be intended for creating a random span mask for infilling, yet the code appears to overwrite it with the padding mask.

### Details:
The code computes a mask from the lengths using:
```python
mask = lens_to_mask(lens, length=seq_len)  # Note: collate_fn already pads to max length in batch
```
This mask ends up being just a padding mask.
A random span mask is then computed:
```python
frac_lengths = torch.zeros((batch,), device=self.device).float().uniform_(*self.frac_lengths_mask)
rand_span_mask = mask_from_frac_lengths(lens, frac_lengths)
```
The following block overwrites the random span mask with the padding mask:
```python
if exists(mask):
    rand_span_mask = mask
```
Since `mask` always exists (due to the padding done by the collate function), `rand_span_mask` is identical to the padding mask.
```python
cond = torch.where(rand_span_mask[..., None], torch.zeros_like(x1), x1)
```
Because `rand_span_mask` is a padding mask, `cond` becomes zero vector.

Could someone clarify the intended purpose of `rand_span_mask`? Is the behavior - overwriting the random span mask with the padding mask - by design?

**Comment by NZqian** [comment]
Actually this is intended. Our model supports taking a latent as input and continuing/editing it through a mask. However, we found that the model tends to copy the given latent segment instead of generating new content. In future work, we will further study the continuation and editing capabilities.

**Comment by romberol** [comment]
Thanks for the clarification!

**Comment by hamigualisingl** [comment]
> Actually this is intended. Our model supports taking a latent as input and continuing/editing it through a mask. However, we found that the model tends to copy the given latent segment instead of generating new content. In future work, we will further study the continuation and editing capabilities.

How can I obtain the latent representation? Could you provide the relevant processing code?
